mailpoet/tests/performance/tests/subscribers-filtering.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-shadow */
+/* eslint-disable import/no-unresolved */
+/* eslint-disable import/no-default-export */
+/**
+ * External dependencies
+ */
+import { sleep, check, group } from 'k6';
+import { chromium } from 'k6/x/browser';
+import { randomIntBetween } from 'https://jslib.k6.io/k6-utils/1.1.0/index.js';
+
+/**
+ * Internal dependencies
+ */
+import {
+  baseURL,
+  thinkTimeMin,
+  thinkTimeMax,
+  headlessSet,
+  adminEmail,
+} from '../config.js';
+import { authenticate } from '../utils/helpers.js';
+/* global Promise */
+
+export function subscribersFiltering() {
+  const browser = chromium.launch({ headless: headlessSet });
+  const page = browser.newPage();
+
+  group('Subscribers - Filter subscribers', function subscribersFiltering() {
+    page
+      .goto(`${baseURL}/wp-admin/admin.php?page=mailpoet-subscribers`, {
+        waitUntil: 'networkidle',
+      })
+
+      .then(() => {
+        authenticate(page);
+      })
+
+      .then(() => {
+        return Promise.all([
+          page.waitForNavigation({ waitUntil: 'networkidle' }),
+        ]);
+      })
+
+      .then(() => {
+        page.locator('[data-automation-id="filters_subscribed"]').click();
+        page.waitForSelector('[data-automation-id="filters_subscribed"]');
+        check(page, {
+          'subscribers filter is visible': page
+            .locator('[data-automation-id="listing_filter_segment"]')
+            .isVisible(),
+        });
+      })
+
+      .then(() => {
+        page
+          .locator('[data-automation-id="listing_filter_segment"]')
+          .selectOption('3');
```

Comment:
Are we concerned about making these tests general enough to run against multiple instances? Hardcoding the filter ID might not be the best way to go if so, but if you have a specific reason for using this filter, or there's no easy way to make it general, this should be fine for now.

Reply:
For some reason it won't work when choosing the list by its name, I will try it once again.
The list `Newsletter mailing list` is always the 3rd by every MailPoet installation, that's why I used it as safe bet for now.

Reply:
Nope, still seeing [this error](https://d.pr/i/9No8fN) when using `({ label: 'Newsletter mailing list (19)' })` and I need to mention how much subscribers are there so the list name is not just `Newsletter mailing list` unfortunately. I'll keep the `3` for now, maybe I have to report a potential bug to k6 for this since it's still beta.

Reply:
Good point about it always being the third list!

I agree it's best to avoid having to know the number of subscribers.

What I was wondering is if there's an easy way to say, "Select the 3rd option in this dropdown". You could maybe try the [press](https://k6.io/docs/javascript-api/k6-browser/api/locator/press/) method to press the down arrow twice and then hit enter to select it? I'm not sure since I haven't been able to get this running myself.

It's also totally fine to leave it as is, especially knowing that `3` should  almost always work for any installation.

Reply:
Yes, for now selecting value 3 as it will be always `Newsletter mailing list` is fine for now until I figure out what's wrong with selecting a label. I have it on my notes of improving existing code after the project ends.

Reply:
FYI, I did some experimentation of my own, and the only way I could find to select the `nth` option in a select list was to first evaluate some javascript to return an array of all the options:

```js
      .then(() => {
        const options = page.evaluate(() => {
          const select = document.querySelector(
            '[data-automation-id="listing_filter_segment"]',
          );
          return Array.from(select.options).map((option) => option.value);
        });
        console.log({options});
        page
          .locator('[data-automation-id="listing_filter_segment"]')
          .selectOption(options[2]);
```

The `options` array looked like this for the test site:
`["","without-list","3","7","8","9","10","11","2","1"]`

So selecting `options[2]` would be equivalent to selecting `'3'`.

This may or may not be helpful!